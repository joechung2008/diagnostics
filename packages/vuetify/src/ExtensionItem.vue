<script setup lang="ts">
import ConfigurationTable from './ConfigurationTable.vue'
import type { ExtensionInfo } from './ExtensionItems.types'
import StageDefinitionTable from './StageDefinitionTable.vue'

defineProps<ExtensionInfo>()
</script>

<template>
  <h1 class="text-h3">
    {{ extensionName }}
  </h1>
  <ConfigurationTable v-if="config" :config="config" />
  <StageDefinitionTable v-if="stageDefinition" :stageDefinition="stageDefinition" />
</template>
