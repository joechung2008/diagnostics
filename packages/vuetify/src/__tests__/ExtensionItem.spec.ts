// ExtensionItem.spec.ts
import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ExtensionItem from '../ExtensionItem.vue'

describe('ExtensionItem', () => {
  it('renders extension name and config', () => {
    const wrapper = mount(ExtensionItem, {
      props: {
        extensionName: 'Test Extension',
        config: {
          foo: 'bar',
          baz: 'qux',
        },
      },
    })
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
