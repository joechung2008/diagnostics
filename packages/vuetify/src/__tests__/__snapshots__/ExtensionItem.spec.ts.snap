// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExtensionItem > renders extension name and config 1`] = `
"<h1 class="text-h3">Test Extension</h1>
<h2 class="text-h4">Configuration</h2>
<div class="v-table v-table--has-top v-table--has-bottom v-theme--light v-table--density-default v-data-table">
  <!---->
  <div class="v-table__wrapper">
    <table>
      <!---->
      <thead>
        <tr>
          <th class="v-data-table__td v-data-table-column--align-start v-data-table__th" colspan="1" rowspan="1">
            <div class="v-data-table-header__content"><span>Key</span>
              <!---->
              <!---->
            </div>
          </th>
          <th class="v-data-table__td v-data-table-column--align-start v-data-table__th" colspan="1" rowspan="1">
            <div class="v-data-table-header__content"><span>Value</span>
              <!---->
              <!---->
            </div>
          </th>
        </tr>
        <!---->
      </thead>
      <!---->
      <tbody>
        <!---->
        <tr class="v-data-table__tr">
          <td class="v-data-table__td v-data-table-column--align-start">foo</td>
          <td class="v-data-table__td v-data-table-column--align-start">bar</td>
        </tr>
        <!---->
        <tr class="v-data-table__tr">
          <td class="v-data-table__td v-data-table-column--align-start">baz</td>
          <td class="v-data-table__td v-data-table-column--align-start">qux</td>
        </tr>
        <!---->
        <!---->
      </tbody>
      <!---->
      <!---->
    </table>
  </div>
  <!---->
</div>
<!--v-if-->"
`;
